featuretools/tests/feature_discovery/test_feature_discovery.py

```diff
@@ -0,0 +1,384 @@
+from typing import Set
+from unittest.mock import patch
+
+import pytest
+from woodwork.column_schema import ColumnSchema
+from woodwork.logical_types import Boolean, Datetime, Double, Ordinal
+
+from featuretools.entityset.entityset import EntitySet
+from featuretools.feature_discovery.feature_discovery import (
+    column_to_keys,
+    feature_to_keys,
+    get_features,
+    get_matching_features,
+    group_features,
+    index_input_set,
+    my_dfs,
+)
+from featuretools.feature_discovery.type_defs import Feature
+from featuretools.primitives import (
+    Absolute,
+    AddNumeric,
+    DateFirstEvent,
+    Day,
+    Equal,
+    Lag,
+    NumUnique,
+    SubtractNumeric,
+)
+from featuretools.synthesis import dfs
+from featuretools.tests.testing_utils.generate_fake_dataframe import (
+    generate_fake_dataframe,
+)
+
+DEFAULT_LT_FOR_TAG = {
+    "category": Ordinal,
+    "numeric": Double,
+    "time_index": Datetime,
+}
+
+
+@pytest.mark.parametrize(
+    "column_schema, expected",
+    [
+        (ColumnSchema(logical_type=Double), ["Double"]),
+        (ColumnSchema(semantic_tags={"index"}), ["index"]),
+        (
+            ColumnSchema(logical_type=Double, semantic_tags={"index", "other"}),
+            ["Double,index", "Double,other"],
+        ),
+    ],
+)
+def test_column_to_keys(column_schema, expected):
+    actual = column_to_keys(column_schema)
+    assert set(actual) == set(expected)
```

Comment:
Nit, but imo it might be cleaner to make the fixture a set so we are not mutating it in the assert

Reply:
I don't think this is mutating, but could be missing something.